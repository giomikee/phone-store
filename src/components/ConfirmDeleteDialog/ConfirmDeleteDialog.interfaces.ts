import type { AddedPhone } from '../../interfaces';

export interface Props {
    phone: AddedPhone;
    onClose: (isDeleteConfirmed: boolean) => void;
}