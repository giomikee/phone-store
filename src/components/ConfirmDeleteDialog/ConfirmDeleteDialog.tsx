import { Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle } from '@mui/material';
import RemoveShoppingCartIcon from '@mui/icons-material/RemoveShoppingCart';
import UndoIcon from '@mui/icons-material/Undo';
import type { Props } from './ConfirmDeleteDialog.interfaces';

const ConfirmDeleteDialog = ({ onClose, phone }: Props) => {
    return (
        <Dialog
            open
            onClose={() => onClose(false)}
            aria-labelledby="alert-dialog-title"
            aria-describedby="alert-dialog-description"
        >
            <DialogTitle id="alert-dialog-title">
                Confirmar acción
            </DialogTitle>
            <DialogContent>
                <DialogContentText id="alert-dialog-description">
                    ¿Eliminar {phone.brand} {phone.name}? 
                </DialogContentText>
            </DialogContent>
            <DialogActions>
                <Button 
                    data-testid='cancel-button'
                    onClick={() => onClose(false)}
                    startIcon={<UndoIcon />}
                    variant='contained'
                >
                    Cancelar
                </Button>
                
                <Button
                    color='error'
                    data-testid='confirm-button'
                    onClick={() => onClose(true)}
                    startIcon={<RemoveShoppingCartIcon />}
                    variant='contained'
                >
                    Confirmar
                </Button>
            </DialogActions>
        </Dialog>
    )
}

export default ConfirmDeleteDialog