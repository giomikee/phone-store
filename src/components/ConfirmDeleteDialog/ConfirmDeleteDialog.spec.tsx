import { describe, expect, it, vi } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/vitest';
import ConfirmDeleteDialog from './ConfirmDeleteDialog';

describe('Given ConfirmDeleteDialog component', () => { 
    const defaultProps = {
        phone: {
            id: 'id',
            name: 'name',
            brand: 'brand',
            basePrice: 0,
            selectedColor: {
                hexCode: 'hexCode',
                name: 'name',
                imageUrl: 'imageUrl'
            },
            selectedStorage: {
                capacity: 'capacity',
                price: 3.1
            },
        },
        onClose: vi.fn(),
    }

    describe('When the component is rendered', () => {
        it('Then it should match the snapshot', () => {
            const { container } = render(<ConfirmDeleteDialog {...defaultProps} />);
            expect(container).toMatchSnapshot();
        });
    });

    describe('When cancel button is clicked', () => {
        it('Then onClose should be called with false', () => {
            render(<ConfirmDeleteDialog {...defaultProps} />);
            fireEvent.click(screen.getByTestId('cancel-button'));
            expect(defaultProps.onClose).toHaveBeenCalledWith(false);
        });
    });
    
    describe('When confirm button is clicked', () => {
        it('Then onClose should be called with false', () => {
            render(<ConfirmDeleteDialog {...defaultProps} />);
            fireEvent.click(screen.getByTestId('confirm-button'));
            expect(defaultProps.onClose).toHaveBeenCalledWith(true);
        });
    });
});