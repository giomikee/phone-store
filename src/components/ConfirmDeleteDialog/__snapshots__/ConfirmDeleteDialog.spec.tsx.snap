// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given ConfirmDeleteDialog component > When the component is rendered > Then it should match the snapshot 1`] = `
<div
  aria-hidden="true"
/>
`;
